velox/functions/sparksql/DecimalVectorFunctions.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/prestosql/ArithmeticImpl.h"
+#include "velox/type/DecimalUtil.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+template <class T>
+class MakeDecimalFunction final : public exec::VectorFunction {
+ public:
+  explicit MakeDecimalFunction(uint8_t precision) : precision_(precision) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args, // Not using const ref so we can reuse args
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& resultRef) const final {
+    VELOX_CHECK_EQ(args.size(), 3);
+    context.ensureWritable(rows, outputType, resultRef);
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto toType = args[1]->type();
+    auto unscaledVec = decodedArgs.at(0);
+    auto nullOnOverflow = decodedArgs.at(2)->valueAt<bool>(0);
+    auto result = resultRef->asUnchecked<FlatVector<T>>()->mutableRawValues();
+    if constexpr (std::is_same_v<T, int64_t>) {
+      rows.applyToSelected([&](int row) {
+        auto unscaled = unscaledVec->valueAt<int64_t>(row);
+        int128_t bound = DecimalUtil::kPowersOfTen[precision_];
+        if (unscaled <= -bound || unscaled >= bound) {
+          // Requested precision is too low to represent this value.
+          if (nullOnOverflow) {
+            resultRef->setNull(row, true);
+          } else {
+            VELOX_USER_FAIL("Unscaled value too large for precision");
```

Comment:
Would you include the 'precision' and the 'value' in this error message to help troubleshooting?